adapter/internal/discovery/xds/callbacks.go

```diff
@@ -45,6 +45,18 @@ func (cb *Callbacks) OnStreamClosed(id int64) {
 // OnStreamRequest prints debug logs
 func (cb *Callbacks) OnStreamRequest(id int64, request *discovery.DiscoveryRequest) error {
 	logger.LoggerXdsCallbacks.Debugf("stream request on stream id: %d Request: %v", id, request)
+	requestEventChannel := GetRequestEventChannel()
+	if "type.googleapis.com/wso2.discovery.api.Api" == request.GetTypeUrl() {
```

Comment:
there You can use `resource.APIType` from pkg `github.com/wso2/adapter/internal/discovery/protocol/resource/v3`

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/1908/commits/31c5c3d62154d211277f27e7ee714fa3b34d632c